### ISSUE TITLE
Calling remove() from slotted element doesn't remove it from parent tree on Firefox

### ISSUE THREAD

**Issue by DiegoCardoso** [webcomponents/webcomponentsjs#863]
Let's suppose I have a small component, which defines a `slot` element appended to its shadow root:
```javascript
class MyComponent extends HTMLElement {
  constructor() {
    super()
    const shadow = this.attachShadow({ mode: 'open'})
    const slot = document.createElement('slot')
    shadow.appendChild(slot)
    console.log('slot created')
  }
}

customElements.define('my-component', MyComponent)
```

And I declare one element like so:

```html
  <my-component id="myComp">
    <p>text</p>
  </my-component>
```

I want to be able to remove the `<p>` element by calling:

```javascript
  const myComp = document.querySelector('#myComp')
  const paragraph = myComp.querySelector('p')
  paragraph.remove()
```

Expected result:
 - Element is removed from the parent tree
 - Calling `myComp.querySelector('p')` returns `null`

Actual result:
 - Element appears to be removed and is not visible
 - Calling `myComp.querySelector('p')` return the same element (and `myComp.querySelector('p').parentNode` points to `myComp`

PS.: If I delete from parent, like
```javascript
  const myComp = document.querySelector('#myComp')
  const paragraph = myComp.querySelector('p')
  myComp.removeChild(paragraph)
```
it works as expected.

Tested on Firefox 53.0.3 and 56.0.2

I have create a small example of the behavior that can be found [here](https://jsfiddle.net/fop2wmrr/2/) (_open DevTools in order to see the logs_).

**Comment by DiegoCardoso** [comment]
Has anyone had an opportunity to take a look into this?

**Comment by TimvdLippe** [comment]
This appears to be fixed: http://jsbin.com/gifidos/edit?html,console,output

**Comment by DiegoCardoso** [comment]
Hi, @TimvdLippe, thanks for taking a look into this issue.

The snippet that I provided is still not working on my tests for Firefox (59), even after updating webcomponentsjs to 1.1.0. 

Now, about the one you provided, I could see as it is working as expected, so I was trying to figure out what is different from both examples. What I found quite odd is that if I surround the part where paragraph is removed with a `setTimeout`, it starts to work as I pointed out on this issue. 
```javascript
setTimeout(() => {
  paragraph.remove();

  console.log(myComp.querySelector('p'));
  
  myComp.removeChild(paragraph)
  
  console.log(myComp.querySelector('p'));
});
```

In fact, this behaviour started to happen even for Chrome (65) and Safari (11.1 and 11.2).

I still can't find the reason for it to happen.

**Comment by TimvdLippe** [comment]
Ah yes confirmed that it breaks with `setTimeout`: http://jsbin.com/gifidos/edit?html,console,output

**Comment by TimvdLippe** [comment]
@DiegoCardoso I was looking through this issue again, but it now seems to be fixed again: http://jsbin.com/wesocekolo/edit?html,console,output Could you confirm/deny?

**Comment by DiegoCardoso** [comment]
@TimvdLippe I've taken a look in the link you provided and in my FF, it seemed to work, but maybe for another reason, because I noticed that the `'connected'` log is being printed after it prints `null`, so it seems that the element is being removed before being fully ready.

If I just add a small timeout to it though, then the paragraph will be logged.

I have changed it a bit in here to display it still with the issue here http://jsbin.com/pilecofado/1/edit?html,console,output

I was able to see the same behavior in Chrome and Safari, so I don't know if this is an issue with webcomponentsjs itself. 

The only difference I noticed from Firefox to other browsers, is that it doesn't work the same way when there's no timeout value.
